core/data/src/test/java/uk/gov/gchq/gaffer/data/elementdefinition/view/ViewTest.java

```diff
@@ -681,6 +682,212 @@ public void shouldReturnFalseWhenViewHasEmptyPostTransformEdgeFilters() {
+                Sets.newHashSet(view.getEntity(TestGroups.ENTITY).getExcludeProperties()));
+    }
+
+    @Test
+    public void shouldOverrideGlobalExcludePropertiesWhenSpecificGroupExcludePropertiesSet() {
+        // Given
+        final View view = new View.Builder()
+                .globalEntities(new GlobalViewElementDefinition.Builder()
+                        .groups(TestGroups.ENTITY)
+                        .excludeProperties(TestPropertyNames.PROP_1, TestPropertyNames.PROP_2)
+                        .build())
+                .entity(TestGroups.ENTITY, new ViewElementDefinition.Builder()
+                        .excludeProperties(TestPropertyNames.PROP_3)
+                        .build())
+                .build();
+
+        // When
+        view.expandGlobalDefinitions();
+
+        // Then
+        assertEquals(Sets.newHashSet(TestPropertyNames.PROP_3),
+                Sets.newHashSet(view.getEntity(TestGroups.ENTITY).getExcludeProperties()));
+    }
+
+    @Test
+    public void shouldThrowExceptionWhenExcludePropertiesAndPropertiesSet() {
+        // Given
+        final View view = new View.Builder()
+                .globalEntities(new GlobalViewElementDefinition.Builder()
+                        .groups(TestGroups.ENTITY)
+                        .excludeProperties(TestPropertyNames.PROP_2)
+                        .build())
+                .entity(TestGroups.ENTITY, new ViewElementDefinition.Builder()
+                        .properties(TestPropertyNames.PROP_2, TestPropertyNames.PROP_3)
+                        .build())
+                .build();
+
+        // When
+        try {
+            view.expandGlobalDefinitions();
+        } catch (IllegalArgumentException e) {
+            assertEquals("You cannot set both properties and excludeProperties", e.getMessage());
+        }
+    }
+
+    @Test
+    public void shouldAddGlobalFiltersToGroup() {
+        // Given
+        final ElementFilter filter = new ElementFilter.Builder()
+                .select(TestPropertyNames.PROP_1)
+                .execute(new Exists())
+                .build();
+
+        final View view = new View.Builder()
+                .globalEntities(new GlobalViewElementDefinition.Builder()
+                        .groups(TestGroups.ENTITY)
+                        .preAggregationFilter(filter)
+                        .build())
+                .entity(TestGroups.ENTITY)
+                .build();
+
+        // When
+        view.expandGlobalDefinitions();
+
+        // Then
+        assertTrue(view.hasPreAggregationFilters());
+        assertEquals(Exists.class.getSimpleName(),
+                view.getEntity(TestGroups.ENTITY).getPreAggregationFilter()
+                        .getComponents().get(0).getPredicate()
+                        .getClass().getSimpleName());
+    }
+
+    @Test
+    public void shouldOverrideGlobalFiltersWhenSpecificGroupFiltersSet() {
```

Comment:
Please could you add similar tests for postAggregationFilter and postTransformFilter.